server/src/main/java/org/elasticsearch/index/mapper/IgnoredSourceFieldMapper.java

```diff
@@ -147,10 +150,37 @@ protected String contentType() {
     @Override
     public void postParse(DocumentParserContext context) {
         // Ignored values are only expected in synthetic mode.
-        assert context.getIgnoredFieldValues().isEmpty() || context.mappingLookup().isSourceSynthetic();
+        if (context.mappingLookup().isSourceSynthetic() == false) {
+            assert context.getIgnoredFieldValues().isEmpty();
+            return;
+        }
+
         List<NameValue> ignoredFieldValues = new ArrayList<>(context.getIgnoredFieldValues());
```

Comment:
Nit: wrap in ArrayList within the branch below, otherwise we don't mutate the returned list.

Reply:
But we use this list outside of the branch?

Reply:
Ah i see what you mean.